x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/detection_rules_client/methods/import_rule.ts

```diff
@@ -34,30 +37,47 @@ export const importRule = async (
     id: undefined,
   });
 
-  if (!existingRule) {
-    const internalRule = convertCreateAPIToInternalSchema(ruleToImport, {
-      immutable: false,
+  if (existingRule && !overwriteRules) {
+    throw createBulkErrorObject({
+      ruleId: existingRule.params.ruleId,
+      statusCode: 409,
+      message: `rule_id: "${existingRule.params.ruleId}" already exists`,
     });
+  }
 
-    return rulesClient.create<RuleParams>({
-      data: internalRule,
-      allowMissingConnectorSecrets,
-    });
-  } else if (existingRule && overwriteRules) {
-    const newInternalRule = convertUpdateAPIToInternalSchema({
+  let importedInternalRule: RuleAlertType;
+
+  if (existingRule && overwriteRules) {
+    const ruleUpdateParams = convertUpdateAPIToInternalSchema({
       existingRule,
       ruleUpdate: ruleToImport,
     });
 
-    return rulesClient.update({
+    importedInternalRule = await rulesClient.update({
       id: existingRule.id,
-      data: newInternalRule,
+      data: ruleUpdateParams,
     });
   } else {
-    throw createBulkErrorObject({
-      ruleId: existingRule.params.ruleId,
-      statusCode: 409,
-      message: `rule_id: "${existingRule.params.ruleId}" already exists`,
+    /* Rule does not exist, so we'll create it */
+    const ruleCreateParams = convertCreateAPIToInternalSchema(ruleToImport, {
+      immutable: false,
+    });
+
+    importedInternalRule = await rulesClient.create<RuleParams>({
+      data: ruleCreateParams,
+      allowMissingConnectorSecrets,
     });
   }
+
+  /* Trying to convert an internal rule to a RuleResponse object */
+  const parseResult = RuleResponse.safeParse(transform(importedInternalRule));
```

Comment:
I think we need to use `internalRuleToAPIResponse` instead of `transform` here. If a rule has an incorrect rule type, the validation will fail at the `RuleResponse` parse stage, but the error would be more cryptic. I suggest reviewing all cases where `transform` is used and replacing them with `internalRuleToAPIResponse`.